src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -13952,7 +13952,7 @@ bool CEEInfo::getTailCallHelpersInternal(CORINFO_RESOLVED_TOKEN* callToken,
 
     TailCallHelp::CreateTailCallHelperStubs(
         m_pMethodBeingCompiled, pTargetMD,
-        msig, isCallvirt, isThisArgByRef,
+        msig, isCallvirt, isThisArgByRef, sig->hasTypeArg(),
```

Comment:
I have seen cases where `pTargetMD->RequiresInstArg()` did not match `sig->hasTypeArg()`. For example, `pTargetMD` is transformed above, but there are other places where it can get out of sync.

`sig->hasTypeArg()` is guaranteed to match what the JIT is going to do.